share/spice/bike_sharing/citi_bike_nyc/bike_sharing_citi_bike_nyc.js

```diff
@@ -22,95 +22,100 @@
+                        pinIcon: 'ddgsi-circle',
+                        pinIconSelected: 'ddgsi-star'
+                    },
+                    model: 'Place',
+                    view: 'Places',
+                    templates: {
+                        item: 'base_item',
+                        options: {
+                            content: Spice.bike_sharing_citi_bike_nyc.content
                         },
-                        sort_fields: {
-                            distance: function(a, b) {
-                                return a.distanceToReference - b.distanceToReference;
-                            }
-                        },
-                        sort_default: 'distance',
-                        data: full_data,
-                        normalize: function(station) {
-                            return {
-                                name: station.name,
-                                lat: station.lat,
-                                lon: station.lon,
-                                distanceToReference: pointOfReference ? L.latLng(station.lat, station.lon).distanceTo(pointOfReference) : station.station_id,
-                                title: station.name,
-                                availableDocks: station.num_docks_available,
-                                availableBikes: station.num_bikes_available,
-                                lastCommunication: moment(new Date(parseInt(station.last_reported)*1000)).fromNow()
-                            };
+                        variants: {
+                            tile: 'narrow'
+                        }
+                    },
+                    sort_fields: {
+                        distance: function(a, b) {
+                            return a.distanceToReference - b.distanceToReference;
                         }
-                    });
+                    },
+                    sort_default: 'distance',
+                    data: full_data,
+                    normalize: function(station) {
+                        return {
+                            name: station.name,
+                            lat: station.lat,
+                            lon: station.lon,
+                            distanceToReference: pointOfReference ? L.latLng(station.lat, station.lon).distanceTo(pointOfReference) : station.station_id,
+                            title: station.name,
+                            availableDocks: station.num_docks_available,
+                            availableBikes: station.num_bikes_available,
+                            lastCommunication: moment(new Date(parseInt(station.last_reported)*1000)).fromNow()
+                        };
+                    }
                 });
-        });
-    }
+            });
+        }
+    };
+    env.ddg_spice_bike_sharing_citi_bike_nyc = bike_sharing_citi_bike.handle_stations_data.bind(bike_sharing_citi_bike);
```

Comment:
This way you don't need to any `bind` and there's no callback context switching. The execution follows as the code is written.